### DISCUSSION TITLE
Difference between this and the built-in ASUSWRT integration

### DISCUSSION THREAD

**Discussion by Valdorama** [Vaskivskyi/ha-asusrouter#170] [Q&A] [answered]
I recently came across this HA integration. I currently use the built-in ASUSWRT integration, and I'm wondering what the difference is between that integration and this one? Thanks.

**Comment by Vaskivskyi** [comment, marked as answer]
Hello, @Valdorama,

Thanks for your interest! 👍🏻 

The first difference is the way of communication with the device. This integration uses the native Asus HTTP(S) API - the same way web panel or mobile app can communicate with routers. In-built ASUSWRT uses either SSH or telnet connection. This leads to all other differences in the available data.


## Both integrations can:
- create `device_tracker`s and show the total number of connected devices via a separate `Connected devices` sensor. But AsusRouter shows lots of useful attributes for the devices, including connection time, RSSI, internet access/connection status and more) - added in 0.4.0
- monitor temperature of CPU and radio modules (this feature is device and firmware dependant) - added in 0.4.0
- load average sensors (but for AsusRouter they are currently limited to the **Merlin firmware only**) - added in 0.4.0


## AsusRouter can:
- Per network interface traffic and speed sensors - selectable by the user and only selected sensors are created to avoid a mess in the list of entities. Includes `WAN` - external traffic with your ISP, `WIRED` (LAN) - local wired traffic, `WLANx` - WiFi traffic for `0` - 2.4 GHz, `1` and `2` - 5 GHz, depending on your router model
![Screenshot 2022-05-15 235248](https://user-images.githubusercontent.com/6554415/168736627-ea4416ce-4e3f-4866-81e7-6ac5ea6cec8c.png)
- Show CPU usage (including per core as attributes)
- Show RAM usage
- Show LAN and WAN ports connection status (including by port as attributes)
![Screenshot 2022-05-17 073427](https://user-images.githubusercontent.com/6554415/168736664-b376902c-6d6a-4170-9590-d66f0c675047.png)
- Boot-time sensor (tells when the last time your device was rebooted)
- WAN IP sensor (+ attributes) - added in 0.3.0
- WAN binary sensor (+ attributes) - added in 0.3.0
- 💡 LED control (on / off) - added in 0.3.0
- 🌐 OpenVPN clients control - added in 0.5.0

## ASUSWRT can:
- Monitor traffic and speed on one Linux network interface (e.g. `eth0` as default) - this could have been useful if it would be working as intended. In reality, mostly it shows some fake upload traffic on different devices (including RT-AC66U, RT-AX88U and others)


## Conclusions
So, they are just different. I've started this AsusRouter integration because ASUSWRT didn't work well for me - showing some non-existing network traffic and strange temperatures.

Also, AsusRouter is still in development and some more features will be added soon, including ones for the device control (e.g. switches for WiFi, services to reboot device, FTP server and so on)


## Updates to this post
- 2022-06-07: added 0.3.0 and 0.4.0 features
- 2022-06-15: added 0.5.0 features

**Comment by Valdorama** [comment]
Thank you for the very detailed reply. This project looks very interesting so I'm going to keep an eye on it and deploy it at some point. Looking forward to the reliable traffic stats (the ASUSWRT data has always been a bit suspect) and the device control (just last week I managed to add switches to HA to block/unblock devices via ssh).

**Comment by rkkoszewski** [comment]
I was just working on a similar integration for HA myself and just came across this project. This should be the default integration for HA, rather than the hacky SSH based one that is used right now upstream. Do you have some kind of roadmap @Vaskivskyi ? I might contribute to this repository with my findings if you accept PRs.

**Comment by Vaskivskyi** [comment]
Hello, @rkkoszewski 

There is no defined roadmap for the integration. At least, not yet.

Out of what I am planning soon, I can list:
- Sensor for external IP (can be helpful for those with dynamic IPs to perform some automation) - implemented in version 0.3.0
- Sensor for external MAC
- Binary sensor of WAN status (connected/disconnected) - implemented in version 0.3.0
- Binary sensors (or switches) for WLANs
- Light entity for LEDs on supported models - implemented in version 0.3.0
- Service to reboot the device

Any other suggestions are welcome and I am accepting PRs from anyone willing to help :slightly_smiling_face:

**Comment by elRadix** [comment]
here is the list if switches/services with their status which I would like to have in your component

https://github.com/jackyaz/scMerlin

**Comment by Vaskivskyi** [comment]
Hello, @elRadix,

Thanks for your feedback! :+1:

I will check it out

**Comment by boheme61** [comment]
Looks "promising" , will check out your great work, as both "ASUSWRT" and "UPnP/IGD" seems like "neglected" Integrations, in terms of what New Routers is capable of.

EDIT: Seems to **work fine with RT-AX68U** , 25 Sensors( i need to go through, to see which i want ) , and so far same amount of Connected-Devices ( tracker ) 👍

Thanks alot, definitely worth to replace asuswrt and  UPnP/IGD with your work, just need the "uptime-sensor" and external IP :)

**Comment by Vaskivskyi** [comment]
> Looks "promising" , will check out your great work, as both "ASUSWRT" and "UPnP/IGD" seems like "neglected" Integrations, in terms of what New Routers is capable of.
> 
> EDIT: Seems to **work fine with RT-AX68U** , 25 Sensors( i need to go through, to see which i want ) , and so far same amount of Connected-Devices ( tracker ) 👍
> 
> Thanks alot, definitely worth to replace asuswrt and UPnP/IGD with your work, just need the "uptime-sensor" and external IP :)

Hello, @boheme61,

Thank you for your feedback! 👍🏻 

The external IP sensor is already implemented in the new version 0.3.0 (as a `wan_ip` entity, not enabled by default, so needs to be enabled manually if you need it).

The uptime sensor is available already as the `boottime`. Since it contains a timestamp of the last router reboot, it has all the needed information (e.g. you can easily calculate uptime using HA templates). The uptime sensor as time since the last reboot will increase the size of the database without any real use (since the new value will be written each time sensor is updated, so every 30 seconds by default)

**Comment by boheme61** [comment]
> The uptime sensor as time since the last reboot will increase the size of the database without any real use (since the new value will be written each time sensor is updated, so every 30 seconds by default)

Good point, never gave that a thought, not that my DB will suffer in regards to Size, thou i do try to minimize any usage i can, And I/O every 30seconds or 1min in this case is as you say "Redundant" , It's more relevant from the "Tracker" , which is the one i use the most

**Comment by Vaskivskyi** [comment]
After adding features of 0.3.0 and 0.4.0 to the comparison list (https://github.com/Vaskivskyi/ha-asusrouter/issues/47#issuecomment-1128442743), seems like AsusRouter is not behind the AsusWRT anymore any sensors.

Of course, development and stability improvement are not over yet 🙂

**Comment by magicse** [comment]
What about Padavan firmware ? Because ASUSWRT work with Padavan firmware throw telnet and ssh well.

**Comment by Vaskivskyi** [comment]
@magicse,

Hello. Unfortunately, I don't have any knowledge about the Padavan FW.

AsusRouter is using HTTP(S) API provided by the stock AsusWRT FW as well as AsusWRT-Merlin FW. If Padavan has any similar API, some of the features of the integration might work. Otherwise, I am afraid, it won't
